### PR TITLE
add app monitoring with okcomputer at /status

### PR DESCRIPTION
**Pull request by ndushay**
connects to #96

### CODE REVIEW

File: config/initializers/okcomputer.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+# Check the memcache servers used by Rails.cache
+if Rails.cache.respond_to? :stats
+  OkComputer::Registry.register 'rails_cache', OkComputer::CacheCheck.new
+else
+  OkComputer::Registry.register 'rails_cache', OkComputer::GenericCacheCheck.new
+end
+
+OkComputer::Registry.register 'stacks_mounted_dir',
+  DirectoryCheck.new(File.join(Settings.stacks.storage_root, 'bb'), read: true, write: true)
```
Comment (tingulfsen) [inline config/initializers/okcomputer.rb]: Is there any reason you're not just using stacks.storage_root here on its own?

File: spec/features/status_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  it '/status checks if Rails app is running' do
+    visit '/status'
+    expect(page.status_code).to eq 200
+    expect(page).to have_text('Application is running')
+  end
+  it '/status/all checks if required dependencies are ok and also shows non-crucial dependencies' do
+    visit '/status/all'
+    expect(page).to have_text('purl_url')
+    expect(page).to have_text('imageserver_url')
+  end
```
Comment (tingulfsen) [inline spec/features/status_spec.rb]: Would you expect a directory check message here also?

File: config/initializers/okcomputer.rb (hunk 1 above)
Comment (jmartin-sul) [inline config/initializers/okcomputer.rb]: maybe just making sure the first part of the druid tree is there for a known druid?  either way, an explanatory comment might be nice.

**Comment by jmartin-sul** [comment]
definitely 👍 once tommy's questions get answered, but👌 overall even as-is.

File: spec/features/status_spec.rb (hunk 2 above)
Comment (ndushay) [inline spec/features/status_spec.rb]: yeah, but i'm really checking for a single required check and for an optional check.  No need to prove that okcomputer works as advertised.

File: config/initializers/okcomputer.rb (hunk 1 above)
Comment (ndushay) [inline config/initializers/okcomputer.rb]: it's referred to as `Settings.stacks.storage_root` in the code.   

```
[ndushay@m3dl-sm-04-mbph digital_stacks_rails (honeybadger-2.6.1)]$ ack storage_root
app/models/stacks_file.rb
36:      File.join(Settings.stacks.storage_root, match[1], match[2], match[3], match[4], file_name) if match && file_name

config/initializers/is_it_working.rb
2:  h.check :directory, path: File.join(Settings.stacks.storage_root, 'bb')

config/settings.yml
2:  storage_root: /stacks

spec/models/stacks_file_spec.rb
8:      expect(subject.path).to eq "#{Settings.stacks.storage_root}/ab/012/cd/3456/def.pdf"

spec/models/stacks_image_spec.rb
88:      expect(subject.path).to eq "#{Settings.stacks.storage_root}/ab/012/cd/3456/def.jp2"
```

Comment (jmartin-sul) [inline config/initializers/okcomputer.rb]: ah, sorry, i'm not sure about tommy, but my question was about the use of `'bb'` in the check (not the use of `Settings` to refer to the setting).  as opposed to just having something like:
```ruby
DirectoryCheck.new(Settings.stacks.storage_root, read: true, write: true)
```

Comment (ndushay) [inline config/initializers/okcomputer.rb]: that's what was in `is_it_working` check.   I assumed it was to ensure that the directory mount really was to the appropriate druid tree level.

File: spec/features/status_spec.rb (hunk 2 above)
Comment (ndushay) [inline spec/features/status_spec.rb]: i could put a comment to that effect at the end of each of those lines if that would be more clear

**Comment by ndushay** [comment]
duly commented per suggestions.

**Pull request merged**